Software/Development/SemesterProject/App/encoderManager.c

```diff
@@ -0,0 +1,195 @@
+
+    //QCLM REGISTER DETERMINES CONSTANT X vs T.
+    EQep1Regs.QEPCTL.bit.QCLM = 1;    // eQEP capture latch mode. 1->latch on unit time out
+    EQep1Regs.QEPCTL.bit.UTE = 1;     // Unit Timeout Enable. 1->Enabled.
+    EQep1Regs.QEPCTL.bit.WDE = 0;     // Watchdog enable. 0->disabled
+
+    EQep1Regs.QPOSMAX = ENCODER_STEPS - 1;         // Maximum possible counter value.
+    EQep1Regs.QPOSINIT = 0;           // Position value used to initialize when index event.
+
+    EQep1Regs.QCAPCTL.bit.CEN = 1;    // QEP Capture Enable
+    EQep1Regs.QCAPCTL.bit.CCPS = 6;   // 1/64 for CAP clock prescaler
+    EQep1Regs.QCAPCTL.bit.UPPS = 5;   // 1/32 for unit position prescaler
+
+    EDIS;
+
+    motorPosSpeedConstructor();
+}
+
+
+/*
+ * Encoder constructor.
+ * To be called as motorPosSpeedConstructor(&motorPosSpeedObject);
+ */
+void motorPosSpeedConstructor(void)
+{
+    motorPosSpeedObject.poles = 2;
+    motorPosSpeedObject.dir = 0;
+    motorPosSpeedObject.thetaMech = 0;
+    motorPosSpeedObject.thetaElec = 0;
+    motorPosSpeedObject.thetaElecOld = 0;
+    motorPosSpeedObject.speedTempCount = 0;
+    motorPosSpeedObject.freqElec = 0;
+    motorPosSpeedObject.freqMech = 0;
+    motorPosSpeedObject.rpmMech = 0;
+    poles_inverse = (float) (1.0 / motorPosSpeedObject.poles);
+}
+
+
+void clearInterruptFlag(void)
+{
+    EQep1Regs.QCLR.bit.UTO=1; // Clear interrupt flag
+}
+
+
+/*
+ * Obtain position and speed of the rotor.
+ * Should be called at switch freq.
+ */
+void posSpeedFromEncoder(void)
+{
+    motorPosCalc();
+    motorSpeedCalc();
+    clearInterruptFlag();
+}
+
+
+/*
+ * Function that updates the object containing position info from encoder.
+ * To be called as motorPosCalc(&motorPosSpeedObject);
+ */
+void motorPosCalc(void)
+{
+    // Check motor direction: 0=CCW/reverse, 1=CW/forward.
+    motorPosSpeedObject.dir = EQep1Regs.QEPSTS.bit.QDF;
+
+    //Update raw angle with counter, remember that QPOSCNT already takes into account direction.
+    motorPosSpeedObject.thetaRaw = EQep1Regs.QPOSCNT;
+
+    // Check an index occurrence
+    if (EQep1Regs.QFLG.bit.IEL == 1) EQep1Regs.QCLR.bit.IEL = 1;   // Clear interrupt flag
+
+    //Transform into electrical angle [�]
+    motorPosSpeedObject.thetaElec = motorPosSpeedObject.thetaRaw * motorPosSpeedObject.poles * REV_TO_RAD * ENCODER_STEPS_INVERSE;
+    if (motorPosSpeedObject.thetaElec >= TWO_PI) motorPosSpeedObject.thetaElec -= TWO_PI;
+    motorPosSpeedObject.thetaMech = motorPosSpeedObject.thetaRaw * REV_TO_RAD * ENCODER_STEPS_INVERSE;
+
+    motorPosSpeedObject.speedTempCount += SW_PERIOD_US;
+}
+
+
+/*
+ *  To ensure the maximum accuracy in the position, the period between two measurements is minimized to switching frequency.
+ *  However, the rotor speed cannot be calculated at the same frequency since it will lead to errors.
+ *  Instead, the speed will be calculated every 10 degrees.
+ */
+void motorSpeedCalc(void)
```

Comment:
I think this function is missing a final capa de pintura for beingg ready: it's too complicated. I think one approach could be adding a short description of the procedure followed in the header and also a bit of separating the code by using downstream functions.